components/supervisor/pkg/supervisor/git.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2020 TypeFox GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package supervisor
+
+import (
+	"context"
+	"time"
+
+	"github.com/gitpod-io/gitpod/supervisor/api"
+	"github.com/gitpod-io/gitpod/supervisor/pkg/gitpod"
+)
+
+// GitTokenProvider provides tokens from git hosting services
+type GitTokenProvider struct {
+	gitpodAPI gitpod.APIInterface
+}
+
+// NewGitTokenProvider creates a new instance of gitTokenProvider
+func NewGitTokenProvider(gitpodAPI gitpod.APIInterface) *GitTokenProvider {
+	return &GitTokenProvider{
+		gitpodAPI: gitpodAPI,
+	}
+}
+
+// GetToken resolves a token from a git hosting service
+func (p *GitTokenProvider) GetToken(ctx context.Context, req *api.GetTokenRequest) (tkn *Token, err error) {
+	token, err := p.gitpodAPI.GetToken(ctx, &gitpod.GetTokenSearchOptions{
+		Host: req.Host,
+	})
+	if err != nil {
+		return nil, err
+	}
+	if token.Value == "" {
+		// token was not found, thus we return just a dummy to satisfy the git protocol
+		return &Token{
+			User:  "oauth2",
```

Comment:
it is not enough to return it here, credential helper get should never fail, but return meaningful output, checked with @AlexTugarev